shimming-toolbox/shimming-toolbox
# Add ability to average across files

## Description
This PR introduces 2 new CLIs:
- `st_image` allows to make modifications on images
   -  Adds the `concat` argument
-  `st_maths` allows to perform math operations on the image
   - Adds the `mean` argument

Only those 2 arguments will be available right now but the CLI framework will be set up to include more math/image operations/manipulations.

st_image concat -h
``` 
Usage: st_image concat [OPTIONS] INPUT...

  Concatenate NIfTIs along the specified dimension

  input: Input paths for the files to concatenate. Separate the files by a
  space.

Options:
  --output PATH       Output filename, supported extensions: .nii, .nii.gz
  --axis [0|1|2|3|4]  Dimension to concatenate  [required]
  --pixdim FLOAT      Pixel resolution to join to image header
  -h, --help          Show this message and exit.
```

st_maths mean -h
```
Usage: st_maths mean [OPTIONS]

  Average data across dimension.

Options:
  --input PATH        Input filename, supported extensions: .nii, .nii.gz
                      [required]

  --output PATH       Output filename, supported extensions: .nii, .nii.gz
  --axis [0|1|2|3|4]  Axis of the array to calculate the average  [required]
  -h, --help          Show this message and exit.
```

## To test the PR 
Make sure `testing_data` is in the root of shimming-toolbox, if not, fix the path to `testing_data`, or use your own files
```
cd shimming-toolbox
pip install -e .
mkdir temp_folder
st_image concat testing_data/realtime_zshimming_data/nifti/sub-example/anat/sub-example_unshimmed_e1.nii.gz testing_data/realtime_zshimming_data/nifti/sub-example/anat/sub-example_unshimmed_e2.nii.gz testing_data/realtime_zshimming_data/nifti/sub-example/anat/sub-example_unshimmed_e3.nii.gz --axis 4 --output temp_folder/concat.nii.gz
st_maths mean --input temp_folder/concat.nii.gz --axis 4 --output temp_folder/mean.nii.gz
```

You should then have a file `temp_folder/mean.nii.gz` which contains the average of the input files

## Linked issues
Fixes #245
